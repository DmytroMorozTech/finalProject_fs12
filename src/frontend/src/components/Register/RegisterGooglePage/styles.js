import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({
  registerPageCard: {
    width: 500,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: '15px'
  },
  form: {
    width: '90%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-around'
  },
  google: {
    width: '90%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    '& > section': {
      width: '100%',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      padding: theme.spacing(1),
      '& > div': {
        flex: 1,
        height: 0.5,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: theme.palette.grey[500]
      },
      '& > p': {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        fontSize: theme.typography.h3.fontSize,
        padding: theme.spacing(1),
        color: theme.palette.grey[500]
      }
    }
  },

  googleBtnWrapper: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },

  googleBtn: {
    border: '2px solid #0a66c2',
    borderRadius: '24px',
    width: '100%',
    fontSize: theme.typography.h3.fontSize,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
    cursor: 'pointer',
    '& > span': {
      marginLeft: 15,
      color: '#0a66c2',
      fontWeight: theme.typography.h5.fontWeight
    },
    '& > img': {
      width: '10%'
    },
    '&:hover': {
      backgroundColor: '#d0e8ff'
    }
  },

  readyLinkedIn: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    fontSize: theme.typography.h5.fontSize
  },

  emailTitle: {
    fontSize: theme.typography.h5.fontSize,
    textAlign: 'center'
  },

  signInLineWrapper: {
    zIndex: 150,
    marginTop: '1rem',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    fontSize: theme.typography.sh3.fontSize
  },

  signInLineLink: {
    textDecoration: 'none',
    fontSize: theme.typography.h3.fontSize,
    fontWeight: theme.typography.sh3.fontWeight,
    lineHeight: '2rem',
    padding: '0.3rem 0.5rem',
    color: '#0a66c2',
    cursor: 'pointer',
    '&:hover': {
      textDecoration: 'underline'
    },
    '&:visited': {
      color: '#0a66c2'
    }
  },

  notYouLineLink: {
    textDecoration: 'none',
    fontSize: theme.typography.h4.fontSize,
    fontWeight: theme.typography.sh3.fontWeight,
    lineHeight: '2rem',
    padding: '0.3rem 0.5rem',
    color: '#0a66c2',
    cursor: 'pointer',
    '&:hover': {
      textDecoration: 'underline'
    },
    '&:visited': {
      color: '#0a66c2'
    }
  },

  joinButton: {
    fontSize: theme.typography.h3.fontSize
  },

  googleSignUpPage: {
    height: 350
  },

  avatarWrapper: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },

  avatar: {
    width: theme.avatar.medium,
    borderRadius: '50%',
    display: 'flex',
    alignSelf: 'flex-start'
  },

  profileButton: {
    borderRadius: '100px',
    height: '1.5rem',
    fontSize: theme.typography.button.fontSize,
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    margin: theme.spacing(1)
  },

  userDataWrapper: {
    width: '18rem',
    marginLeft: theme.spacing(2),
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column'
  },

  userName: {
    whiteSpace: 'normal',
    fontSize: theme.typography.sh3.fontSize,
    fontWeight: theme.typography.h6.fontWeight
  },

  userProfession: {
    width: '100%',
    whiteSpace: 'normal',
    fontSize: theme.typography.h6.fontSize
  }
}))