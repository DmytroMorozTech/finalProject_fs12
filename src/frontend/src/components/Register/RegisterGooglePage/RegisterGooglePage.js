import React from 'react'
import styles from './styles'
import SharedButton from '../../../shared/SharedButton/SharedButton'
import {TextField} from 'formik-material-ui'
import {Paper} from '@material-ui/core'
import http from '../../../services/httpService'
import {toast} from 'react-toastify'
import {useHistory} from 'react-router'
import {Field, Form, Formik} from 'formik'
import * as Yup from 'yup'
import clsx from 'clsx'
import {Link} from 'react-router-dom'
import {useDispatch, useSelector} from 'react-redux'
import {activeUserSelector} from '../../../redux/User/userSelector'
import Image from '../../../shared/Image/Image'
import {signOutAction} from '../../../redux/User/userActions'

const RegisterGooglePage = () => {
  const classes = styles()
  const activeUser = useSelector(activeUserSelector)
  const history = useHistory()
  const dispatch = useDispatch()

  const handleSubmit = (data) => {
    const {password} = data

    http
      .post('api/update_password', {
        email: activeUser.email,
        password: password
      })
      .then(res => {
        if (res.status === 200) {
          toast.info('You have signed up successful! Now you can sign in with your email and password! ')
          history.push('/home')
        }
      })
      .catch(err => {
        const errorMsg = err.response.data.message
        toast.error(errorMsg)
      })
  }
  return (
    <Paper elevation={3} className={clsx(classes.registerPageCard, classes.googleSignUpPage)}>
      <div className={classes.avatarWrapper}>
        <Image
          imageUrl={activeUser.avatarPublicId}
          alt={`Avatar of ${activeUser.fullName}`}
          className={classes.avatar}
          type={'smallAvatar'}
        />
        <div className={classes.userDataWrapper}>
          <span className={classes.userName}>{activeUser.fullName}</span>
          <span className={classes.userName}>{activeUser.email}</span>
        </div>
        <Link to="/" className={classes.notYouLineLink} onClick={() => dispatch(signOutAction())}>Not you?</Link>
      </div>
      <Formik
        initialValues={{
          password: ''
        }}
        validationSchema={
          Yup.object().shape({
            password: Yup.string()
              .matches('^(?=.*[A-Za-z])(?=.*\\d)(?=.*[@$!%*#?&])[A-Za-z\\d@$!%*#?&]{8,}$', 'Password must be: minimum eight characters, at least one letter, one number and one special character!')
              .required('Password is required')
          })}
        onSubmit={(values, {resetForm}) => {
          handleSubmit(values)
          resetForm({password: ''})
        }}
      >
        {() => (
          <Form className={classes.form}>
            <Field component={TextField}
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="password"
              label="Password (8 or more characters)"
              type="password"
            />
            <SharedButton
              className={classes.joinButton}
              type="submit"
              size="large"
              title="Join"
              fullWidth
              variant="contained"
              color="primary"
            />
          </Form>
        )}
      </Formik>
      <div className={classes.signInLineWrapper}>
        <p>Already on LinkedIn?</p>
        <Link to="/" className={classes.signInLineLink} onClick={() => dispatch(signOutAction())}>Sign in</Link>
      </div>
    </Paper>
  )
}

export default RegisterGooglePage