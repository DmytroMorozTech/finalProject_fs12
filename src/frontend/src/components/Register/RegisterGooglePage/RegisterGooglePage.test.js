import { shallow } from 'enzyme'
import React from 'react'
import RegisterGooglePage from './RegisterGooglePage'
import * as redux from 'react-redux'

jest.mock('./styles', () => () => ({}))

const setUp = () => shallow(<RegisterGooglePage />)

describe('should render RegisterGooglePage component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let component

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(false)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp()
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  it('should work dispatch sign out', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('Link[id=\'signOut\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
  it('should work dispatch sign in', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('Link[id=\'signIn\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})
