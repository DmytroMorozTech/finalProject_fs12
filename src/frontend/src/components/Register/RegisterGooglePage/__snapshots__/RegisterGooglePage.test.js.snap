// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RegisterGooglePage component to match snapshot 1`] = `
<WithStyles(ForwardRef(Paper))
  className=""
  elevation={3}
>
  <div>
    <Image
      alt="Avatar of undefined"
      type="smallAvatar"
    />
    <div>
      <span />
      <span />
    </div>
    <Link
      id="signOut"
      onClick={[Function]}
      to="/"
    >
      Not you?
    </Link>
  </div>
  <Formik
    initialValues={
      Object {
        "password": "",
      }
    }
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_blacklistError": undefined,
        "_excludedEdges": Array [],
        "_nodes": Array [
          "password",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_whitelistError": undefined,
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "password": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "matches": false,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <Component />
  </Formik>
  <div>
    <p>
      Already on LinkedIn?
    </p>
    <Link
      id="signIn"
      onClick={[Function]}
      to="/"
    >
      Sign in
    </Link>
  </div>
</WithStyles(ForwardRef(Paper))>
`;
